snuba/web/views.py

```diff
@@ -134,9 +136,18 @@ def check_clickhouse(
         entities = itertools.chain(
             *[dataset.get_all_entities() for dataset in datasets]
         )
-        storages = list(
-            itertools.chain(*[entity.get_all_storages() for entity in entities])
-        )
+
+        storages = []
+        for entity in entities:
+            entity_storages = entity.get_all_storages()
+            for storage in entity_storages:
+                assert isinstance(storage, ReadableTableStorage)
```

Comment:
Why does the storage need to be `ReadableTableStorage`? Mind providing a bit more context?

Reply:
The type checking is used to access the `get_storage_key()` method which is not implemented on the base Storage class.